tsfresh/feature_dynamics_extraction/feature_dynamics_utils.py

```diff
@@ -0,0 +1,536 @@
+            )
+        )
+
+    return {"window_length": window_length, "fts_parts": fts_parts}
+
+
+def parse_feature_dynamics_parts(full_feature_name):
+    """
+    Generates the feature dynamics (fd) parts from a full feature dynamic (fd) name
+
+    i.e. given a full feature dynamic name like:
+    <ts_kind>||<fts_name>||<fts_params>@<window_length>__<fd_name>__<fd_params>
+
+    Then this function will return a dictionary with one key:
+    1) fd_parts, a list containing:
+    [
+        <ts_kind>||<fts_name>||<fts_params>@<window_length>,
+        <fd_name>,
+        <fd_params>*
+    ]
+    (where multiple fd_params are allowed and are each a separate element)
+    """
+    # Split according to our separator into <col_name>, <feature_name>, <feature_params>
+    fd_parts = full_feature_name.split("__")
+    n_fd_parts = len(fd_parts)
+    if n_fd_parts == 1:
+        raise ValueError(
+            "Splitting of columnname {} resulted in only one part.".format(
+                full_feature_name
+            )
+        )
+    return {"fd_parts": fd_parts}
+
+
+def derive_features_dictionaries(feature_names: List[str]) -> Tuple[dict, dict]:
+    """
+    Derives and writes out two feature dictionaries which can be used with the feature dynamics framework.
+
+        params:
+            feature_names (list of str): the relevant feature names in the form of:
+            <ts_kind>||<feature_time_series>||<feature_times_series_params>@<window_length>__<feature_dynamic_name>__<feature_dynamics_params>
+
+        returns:
+            fts_mapping (dict): The mapping used to compute the feature time-series on the ts kinds.
+            fd_mapping (dict): The mapping used to compute the feature dynamics on the feature time series.
+
+            The fts and fd mappings map window_lengths to a dictionary of column names (either
+            ts_kinds names or fts names) which map to a dictionary of feature calculators.
+            i.e. window_length ---> column_name ---> feature_calculator_name ---> [*feature_calculator_params]
+
+    """
+    fts_mapping = {}
+    fd_mapping = {}
+
+    for full_feature_name in feature_names:
+        if not isinstance(full_feature_name, str):
+            raise TypeError(
+                "Column name {} should be a string or unicode".format(full_feature_name)
+            )
+
+        fts_information = parse_feature_timeseries_parts(full_feature_name)
+        fts_mapping = update_feature_dictionary(
+            fts_mapping,
+            window_length=fts_information.get("window_length"),
+            feature_parts=fts_information.get("fts_parts"),
+        )
+
+        fd_information = parse_feature_dynamics_parts(full_feature_name)
+        fd_mapping = update_feature_dictionary(
+            fd_mapping,
+            window_length=fts_information.get("window_length"),
+            feature_parts=fd_information.get("fd_parts"),
+        )
+
+    return fts_mapping, fd_mapping
+
+
+def diff_within_series(
```

Comment:
@kempa-liehr worth noting that the different input formats and lead to different _ordering_ of time series differences. For example, `ts_kind_1 - ts_kind_2` (labelled as `D_ts_kind_1_ts_kind_2`) when a wide df format is passed in and  `ts_kind_2 - ts_kind_1` (labelled as `D_ts_kind_2_ts_kind_1`) when a long df format is passed in. 

This can be fixed if we think it is necessary to have consistent differences.